src/tooling/piral-cli/src/injectors/pilet.ts

```diff
@@ -15,18 +18,27 @@ export interface PiletInjectorConfig extends KrasInjectorConfig {
   pilets: Array<Pilet>;
   api: string;
   app: string;
+  feed?: string;
+}
+
+interface PiletMetaData {
+  name?: string
+  [key: string]: unknown;
 }
 
 export default class PiletInjector implements KrasInjector {
   public config: PiletInjectorConfig;
   private piletApi: string;
+  private remotePiletsPromise: Promise<PiletMetaData[]>;
 
   constructor(options: PiletInjectorConfig, config: KrasConfiguration, core: EventEmitter) {
     this.config = options;
     // either take a full URI or make it an absolute path relative to the current origin
     this.piletApi = /^https?:/.test(options.api)
       ? options.api
       : `${config.ssl ? 'https' : 'http'}://localhost:${config.port}${options.api}`;
+
+    this.remotePiletsPromise = this.loadRemoteFeed(this.config.feed);
```

Comment:
Should this be loaded only once or always when a feed requested?

Reason I am asking is because in the current impl. any change to the (remote) feed would require a restart of the dev server / debug process.

Reply:
hmm... right now it would only be loaded once and a restart would be required, yes.
Mainly I was trying to save some performance and considered the remote feed service to be kinda stable anyway - at least more stable than the `pilet debug` instance.
But maybe performance isn't really too big of a concern here so maybe it would be better to have it always pull the latest data from the remote feed. What do you think?

Reply:
Yeah so I see four options here:

1. Only load in the beginning (keep as is)
2. Load in the beginning and refresh in a certain interval (e.g., 5 min) -> report in the console when the remote feed data was refreshed (info level)
3. Load in the beginning and refresh upon request if the last request is too old (e.g., older than 1 minute); like (2) but lazy -> easiest way to achieve this would be to have 1 min timer that resets the current result such that a refetch is necessary; I think it could be even done on Axios such that we "always" fetch, but tell Axios that we want to cache results for 1 minute (or whatever time)
4. Make the request on every call

I think (4) is the easiest implementation-wise, but could be the worst from UX (considering happy case = stable feed, we need fast response times). Maybe going for (3) in a non-blocking way (i.e., do not throw away the result if we exceed a time, but just refetch while serving the previous / last result).

Reply:
One disadvantage of the non-blocking refresh would be, that the developer would need to refresh twice to see the changed feed, which could lead to some confusion.

In general, the `$pilet-api` feed is anyway only called relatively infrequently I think. Since HMR is active for the local pilets, a lot of the time developers don't need to do a full page refresh. So a reload of the `$pilet-api` is - at least from my experience - relatively rare.

So maybe waiting a little bit more for that request to come back but avoiding confusion about when potential changes on the remote feed would actually be applied could be the better compromise?

Reply:
> One disadvantage of the non-blocking refresh would be, that the developer would need to refresh twice to see the changed feed, which could lead to some confusion.

True.

> So maybe waiting a little bit more for that request to come back but avoiding confusion about when potential changes on the remote feed would actually be applied could be the better compromise?

Yeah sounds great! Thanks!

Reply:
pushed the changes
the remote feed is now always loaded fresh when a request to the pilet feed is sent

Reply:
Awesome - thanks! Great job 🥇